templates/model/spec.go

```diff
@@ -288,3 +297,118 @@ func (i *Include) Validate() error {
 		nonEmptySlice(i.Pos, i.Paths, "paths"),
 	)
 }
+
+// RegexReplace is an action that replaces a regex match (or a subgroup of it) with a
+// template expression.
+type RegexReplace struct {
+	// Pos is the YAML file location where this object started.
+	Pos *ConfigPos `yaml:"-"`
+
+	Regex    String `yaml:"regex"`
+	Subgroup Int    `yaml:"subgroup"`
+	With     String `yaml:"with"`
```

Comment:
Comments (throughout)? Given this file is the source of truth of the spec.

Also, should we add go example later?

Reply:
Yeah, we should do that. Let's come back and do documentation once we get something running.